library/ssl_tls.c

```diff
@@ -5957,36 +5957,53 @@ void mbedtls_ssl_conf_curves( mbedtls_ssl_config *conf,
 {
     conf->curve_list = curve_list;
 }
-#endif
+#endif /* MBEDTLS_ECP_C */
 
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
 int mbedtls_ssl_set_hostname( mbedtls_ssl_context *ssl, const char *hostname )
 {
-    size_t hostname_len;
+    /* Initialize to suppress unnecessary compiler warning */
+    size_t hostname_len = 0; 
 
-    if( hostname == NULL )
-        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+    /* Check if new hostname is valid before making any change to current one */
 
-    hostname_len = strlen( hostname );
+    if( hostname != NULL )
+    {
+	hostname_len = strlen( hostname );
 
-    if( hostname_len + 1 == 0 )
-        return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
+	if( hostname_len > MBEDTLS_SSL_MAX_HOST_NAME_LEN )
+	    return( MBEDTLS_ERR_SSL_BAD_INPUT_DATA );
```

Comment:
Ah I see what you mean. Its unlikely because the RFC defines `MBEDTLS_SSL_MAX_HOST_NAME_LEN` as a 255, I thought that was a user configurable. Also, I do not think this will be a problem in machines with smaller word size than 32